src/java/boa/compiler/visitors/CodeGeneratingVisitor.java

```diff
@@ -175,8 +178,14 @@ public void visit(final FixPExpression n) {
 		/** {@inheritDoc} */
 		@Override
 		public void visit(final VarDeclStatement n) {
-			if (n.type instanceof BoaTable)
+			if (n.type instanceof BoaTable) {
+				UserDefinedAgg.Builder function = UserDefinedAggregators.findByUserGivenName(((OutputType)n.getType()).getId().getToken());
+				if(function != null) {
```

Comment:
seems odd to set this here - shouldnt it already be set when it is found and returned?